plasmapy/dispersion/numerical/hollweg_.py

```diff
@@ -0,0 +1,352 @@
+            f"Argument 'k' needs to be single valued or a 1D array astropy Quantity,"
+            f" got array of shape {k.shape}."
+        )
+    if np.any(k <= 0):
+        raise ValueError("Argument 'k' can not be a or have negative values.")
+
+    # validate argument theta
+    theta = theta.squeeze()
+    if not (theta.ndim == 0 or theta.ndim == 1):
+        raise ValueError(
+            f"Argument 'theta' needs to be a single valued or 1D array astropy "
+            f"Quantity, got array of shape {k.shape}."
+        )
+
+    # Single k value case
+    if np.isscalar(k.value):
+        k = (
+            np.array(
+                [
+                    k.value,
+                ]
+            )
+            * u.rad
+            / u.m
+        )
+
+    # Calc needed plasma parameters
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=PhysicsWarning)
+        n_e = z_mean * n_i
+        c_s = pfp.ion_sound_speed(
+            T_e=T_e,
+            T_i=T_i,
+            ion=ion,
+            n_e=n_e,
+            gamma_e=gamma_e,
+            gamma_i=gamma_i,
+            z_mean=z_mean,
+        ).value
+        v_A = pfp.Alfven_speed(B, n_i, ion=ion, z_mean=z_mean).value
+        omega_ci = pfp.gyrofrequency(B=B, particle=ion, signed=False, Z=z_mean).value
+        omega_pe = pfp.plasma_frequency(n=n_e, particle="e-").value
+
+    # strip units from select input args
+    k = k.value
+    theta = theta.value
+    cs_vA = c_s / v_A
+
+    # Parameters kx and kz
+    kz = np.cos(theta) * k
+    kx = np.sin(theta) * k
+
+    # Define helpful parameters
+    beta = (c_s / v_A) ** 2
+    alpha_A = (k * v_A) ** 2
+    alpha_s = (k * c_s) ** 2  # == alpha_A * beta
+    sigma = (kz * v_A) ** 2
+    D = (c_s / omega_ci) ** 2
+    F = (c.value / omega_pe) ** 2
+
+    # Polynomial coefficients: c3*x^3 + c2*x^2 + c1*x + c0 = 0
+    c3 = F * kx ** 2 + 1
+    c2 = -alpha_A * (1 + beta + F * kx ** 2) - sigma * (1 + D * kx ** 2)
+    c1 = sigma * alpha_A * (1 + 2 * beta + D * kx ** 2)
+    c0 = -alpha_s * sigma ** 2
+
+    # Find roots to polynomial
+    coefficients = np.array([c3, c2, c1, c0], ndmin=2)
+    nroots = coefficients.shape[0] - 1  # 3
+    nks = coefficients.shape[1]
+    roots = np.empty((nroots, nks), dtype=np.complex128)
+    for ii in range(nks):
+        roots[:, ii] = np.roots(coefficients[:, ii])[:]
+
+    roots = np.sqrt(roots)
+    roots = np.sort(roots, axis=0)
+
+    # Warn about NOT low-beta
+    if c_s / v_A > 0.1:
+        warnings.warn(
+            f"This solver is valid in the low-beta regime, "
+            f"c_s/v_A << 1 according to Bellan, 2012, Sec. 1.7 "
+            f"(see documentation for DOI). A c_s/v_A value of {cs_vA:.2f} "
+            f"was calculated which may affect the validity of the solution.",
+            PhysicsWarning,
+        )
+
+    # Warn about theta not nearly perpendicular
+    if theta < 0.5 * np.pi - 0.1 or theta > 0.5 * np.pi + 0.1:
```

Comment:
```suggestion
    if np.abs(theta - np.pi / 2) > 0.1:
```

Just for improved readability.